Flaky test org.eclipse.jetty.quic.quiche.jna.LowLevelQuicheClientCertTest.testClientCert

```
[ERROR] org.eclipse.jetty.quic.quiche.jna.LowLevelQuicheClientCertTest.testClientCert  Time elapsed: 0.324 s  <<< FAILURE!
java.lang.AssertionError: 

Expected: is <72>
     but: was <71>
	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:6)
	at org.eclipse.jetty.quic.quiche.jna@12.0.0-SNAPSHOT/org.eclipse.jetty.quic.quiche.jna.LowLevelQuicheClientCertTest.drainServerToFeedClient(LowLevelQuicheClientCertTest.java:144)
	at org.eclipse.jetty.quic.quiche.jna@12.0.0-SNAPSHOT/org.eclipse.jetty.quic.quiche.jna.LowLevelQuicheClientCertTest.connectClientToServer(LowLevelQuicheClientCertTest.java:220)
	at org.eclipse.jetty.quic.quiche.jna@12.0.0-SNAPSHOT/org.eclipse.jetty.quic.quiche.jna.LowLevelQuicheClientCertTest.testClientCert(LowLevelQuicheClientCertTest.java:122)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)

```

--- a/jetty-quic/quic-quiche/quic-quiche-foreign-incubator/src/test/java/org/eclipse/jetty/quic/quiche/foreign/incubator/LowLevelQuicheClientCertTest.java
+++ b/jetty-quic/quic-quiche/quic-quiche-foreign-incubator/src/test/java/org/eclipse/jetty/quic/quiche/foreign/incubator/LowLevelQuicheClientCertTest.java
@@ -39,6 +39,9 @@ import org.junit.jupiter.api.extension.ExtendWith;
 
 import static org.eclipse.jetty.quic.quiche.Quiche.QUICHE_MIN_CLIENT_INITIAL_LEN;
 import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.both;
+import static org.hamcrest.Matchers.greaterThanOrEqualTo;
+import static org.hamcrest.Matchers.lessThanOrEqualTo;
 import static org.hamcrest.Matchers.not;
 import static org.hamcrest.Matchers.nullValue;
 import static org.hamcrest.core.Is.is;
@@ -148,6 +151,20 @@ public class LowLevelQuicheClientCertTest
         assertThat(fed, is(expectedSize));
     }
 
+    private void drainServerToFeedClient(Map.Entry<ForeignIncubatorQuicheConnection, ForeignIncubatorQuicheConnection> entry, int expectedSizeLowerBound, int expectedSizeUpperBound) throws IOException
+    {
+        ForeignIncubatorQuicheConnection clientQuicheConnection = entry.getKey();
+        ForeignIncubatorQuicheConnection serverQuicheConnection = entry.getValue();
+        ByteBuffer buffer = ByteBuffer.allocate(QUICHE_MIN_CLIENT_INITIAL_LEN);
+
+        int drained = serverQuicheConnection.drainCipherBytes(buffer);
+
+        assertThat(drained, is(both(greaterThanOrEqualTo(expectedSizeLowerBound)).and(lessThanOrEqualTo(expectedSizeUpperBound))));
+        buffer.flip();
+        int fed = clientQuicheConnection.feedCipherBytes(buffer, clientSocketAddress, serverSocketAddress);
+        assertThat(fed, is(both(greaterThanOrEqualTo(expectedSizeLowerBound)).and(lessThanOrEqualTo(expectedSizeUpperBound))));
+    }
+
     private void drainClientToFeedServer(Map.Entry<ForeignIncubatorQuicheConnection, ForeignIncubatorQuicheConnection> entry, int expectedSize) throws IOException
     {
         ForeignIncubatorQuicheConnection clientQuicheConnection = entry.getKey();
@@ -218,7 +235,7 @@ public class LowLevelQuicheClientCertTest
         assertThat(clientQuicheConnection.isConnectionEstablished(), is(true));
 
         // 2nd round (needed b/c of client cert)
-        drainServerToFeedClient(entry, 71);
+        drainServerToFeedClient(entry, 71, 72);
         assertThat(serverQuicheConnection.isConnectionEstablished(), is(false));
         assertThat(clientQuicheConnection.isConnectionEstablished(), is(true));
 

--- a/jetty-quic/quic-quiche/quic-quiche-jna/src/test/java/org/eclipse/jetty/quic/quiche/jna/LowLevelQuicheClientCertTest.java
+++ b/jetty-quic/quic-quiche/quic-quiche-jna/src/test/java/org/eclipse/jetty/quic/quiche/jna/LowLevelQuicheClientCertTest.java
@@ -38,6 +38,9 @@ import org.junit.jupiter.api.extension.ExtendWith;
 
 import static org.eclipse.jetty.quic.quiche.Quiche.QUICHE_MIN_CLIENT_INITIAL_LEN;
 import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.both;
+import static org.hamcrest.Matchers.greaterThanOrEqualTo;
+import static org.hamcrest.Matchers.lessThanOrEqualTo;
 import static org.hamcrest.Matchers.not;
 import static org.hamcrest.Matchers.nullValue;
 import static org.hamcrest.core.Is.is;
@@ -147,6 +150,20 @@ public class LowLevelQuicheClientCertTest
         assertThat(fed, is(expectedSize));
     }
 
+    private void drainServerToFeedClient(Map.Entry<JnaQuicheConnection, JnaQuicheConnection> entry, int expectedSizeLowerBound, int expectedSizeUpperBound) throws IOException
+    {
+        JnaQuicheConnection clientQuicheConnection = entry.getKey();
+        JnaQuicheConnection serverQuicheConnection = entry.getValue();
+        ByteBuffer buffer = ByteBuffer.allocate(QUICHE_MIN_CLIENT_INITIAL_LEN);
+
+        int drained = serverQuicheConnection.drainCipherBytes(buffer);
+
+        assertThat(drained, is(both(greaterThanOrEqualTo(expectedSizeLowerBound)).and(lessThanOrEqualTo(expectedSizeUpperBound))));
+        buffer.flip();
+        int fed = clientQuicheConnection.feedCipherBytes(buffer, clientSocketAddress, serverSocketAddress);
+        assertThat(fed, is(both(greaterThanOrEqualTo(expectedSizeLowerBound)).and(lessThanOrEqualTo(expectedSizeUpperBound))));
+    }
+
     private void drainClientToFeedServer(Map.Entry<JnaQuicheConnection, JnaQuicheConnection> entry, int expectedSize) throws IOException
     {
         JnaQuicheConnection clientQuicheConnection = entry.getKey();
@@ -217,7 +234,7 @@ public class LowLevelQuicheClientCertTest
         assertThat(clientQuicheConnection.isConnectionEstablished(), is(true));
 
         // 2nd round (needed b/c of client cert)
-        drainServerToFeedClient(entry, 72);
+        drainServerToFeedClient(entry, 71, 72);
         assertThat(serverQuicheConnection.isConnectionEstablished(), is(false));
         assertThat(clientQuicheConnection.isConnectionEstablished(), is(true));
 

